plugins/woocommerce-blocks/assets/js/blocks/product-gallery/edit.tsx

```diff
@@ -136,16 +132,14 @@ export const Edit = ( {
 
 	useEffect( () => {
 		const mode = getMode( currentTemplateId, templateType );
-		const newProductGalleryClientId =
-			attributes.productGalleryClientId || clientId;
 
 		setAttributes( {
 			...attributes,
 			mode,
-			productGalleryClientId: newProductGalleryClientId,
+			productGalleryClientId: clientId,
 		} );
 		// Move the Thumbnails block to the correct above or below the Large Image based on the thumbnailsPosition attribute.
-		moveInnerBlocksToPosition( attributes, newProductGalleryClientId );
+		moveInnerBlocksToPosition( attributes, clientId );
```

Comment:
It may be me misunderstanding the concept of using `clientId` and  `productGalleryClientId` attribute. But my feeling is we should use attribute here. Also, in line 158 raw `clientId` is passed to the context. 

```suggestion
		moveInnerBlocksToPosition( attributes ); // and just read the `productGalleryClientId` from `attributes` inside the function
```

Otherwise seems the whole `productGalleryClientId` attribute is redundant as all the blocks inside seems to be relying on `context.productGalleryClientId` 🤔